ocpmodels/trainers/base_trainer.py

```diff
@@ -60,9 +61,10 @@ def load_config_from_yaml_and_cmd(self, args):
         self.config = build_config(args)
 
         # device
-        self.device = torch.device(
-            "cuda" if torch.cuda.is_available() else "cpu"
-        )
+        if torch.cuda.is_available():
+            self.device = self.config["optim"].get("output_device", 0)
```

Comment:
Hmm, I might be missing something here; but aren't these automatically inferred from `CUDA_VISIBLE_DEVICES`?

Reply:
So this is a bit tricky. How this was written before used `cuda` as the device, which then checks the environment variable and uses that. But we need to specify a specify a device number for the master device, and couldn't find anything specifically to tell you what device cuda defaults. We could have `device_ids` be equivalent to the devices `CUDA_VISIBLE_DEVICES` shows, and then have master be the 1st device here?

Reply:
I see. Yeah, defaulting to `CUDA_VISIBLE_DEVICES` and setting 1st device as master sounds good.